server/src/main/java/org/elasticsearch/indices/SystemIndexDescriptor.java

```diff
@@ -368,6 +449,16 @@ public Builder setMinimumNodeVersion(Version version) {
             return this;
         }
 
+        public Builder setType(Type type) {
+            this.type = type;
+            return this;
+        }
+
+        public Builder setAllowedStackComponents(List<String> allowedStackComponents) {
```

Comment:
Nit: It's `allowedStackComponents` here in the builder and `allowedElasticProductOrigins` elsewhere, we should probably be consistent.